### PR TITLE
WIP: metadata docs

### PR DESCRIPTION
**Pull request by iktl**
Add metadata docs generated between Markdown and introspection queries against the API.

### CODE REVIEW

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Metadata API Overview"
+id: "metadata-overview"
+---
+
+Every time that dbt Cloud runs a dbt project, it generates metadata which pertains to the accuracy, recency, configuration, and structure of the views and tables in the warehouse. dbt Cloud serves a GraphQL API which supports arbitrary queries over this metadata; the endpoint for this API is `https://metadata.cloud.getdbt.com/graphql`. This API is an incredibly rich resource for evaluating data health longitudinally or at a point in time. Please note that today, any query on our metadata API today will refer to the last run.
```
Comment (ashnm) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md:6]: we're missing an i in this url, should be:
https://metadata.cloud.getdbt.com/graphiql

File: website/docs/docs/dbt-cloud/dbt-cloud-api/admin-cloud-api.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+The dbt Cloud Administrative API is enabled by default for _Team_ and _Enterprise_ plans. It can be used to:
+
+- Download artifacts after a job has completed
+- Kick off a job run from an orchestration tool
+- Manage your dbt Cloud account
+- and more
+
+You can learn more information [here](/dbt-cloud/api).
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/admin-cloud-api.md]:
```suggestion
You can find reference documentation for the dbt Cloud Administrative API [here](/dbt-cloud/api).
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/cloud-apis.md [hunk 3]
```diff
@@ -1,33 +1,18 @@
...
 ---
 
 ## Overview
 
 Accounts on the _Team_ and _Enterprise_ plans can query the dbt Cloud APIs.
 
-dbt Cloud provides two APIs.  
+dbt Cloud provides two APIs.
 
 The [dbt Cloud Administrative API](/dbt-cloud/api) can be used to administrate a dbt Cloud account.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/cloud-apis.md]: can we link to the `/docs/dbt-cloud/dbt-cloud-api/admin-cloud-api` page here?

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+title: "Metadata API Overview"
+id: "metadata-overview"
+---
+
+Every time that dbt Cloud runs a dbt project, it generates metadata which pertains to the accuracy, recency, configuration, and structure of the views and tables in the warehouse. dbt Cloud serves a GraphQL API which supports arbitrary queries over this metadata; the endpoint for this API is `https://metadata.cloud.getdbt.com/graphql`. This API is an incredibly rich resource for evaluating data health longitudinally or at a point in time.
+
+We also provide a [graphical explorer](https://metadata.cloud.getdbt.com/graphiql) for this API where you can run ad-hoc queries or browse the schema. As GraphQL provides a self-describing API, the schema shown in the GraphiQL interface is an accurate representation of the graph and fields available to query on at any point in time.
+
+The metadata API is available to accounts on the _Team_ and _Enterprise_ plans, for any version >= dbt v0.19.0.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md]:
```suggestion
The metadata API is available to accounts on the _Team_ and _Enterprise_ plans, for any version >= dbt v0.19.0. Please note that artifacts generated with a version of dbt _less than_ v0.19.0 will not be accessible via the dbt Cloud Metadata API.
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+*Please note that today, any query on our metadata API today will refer to the last run. We will continuously expand the capabilities of the metadata API, and are open to all feedback and suggestions at metadata@dbtlabs.com.*
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview.md]:
```suggestion
*Please note that at present, queries to the Metadata API exclusively return metadata about the most recent Run for a given dbt Cloud Job. We are continuously expanding the capabilities of the metadata API and we welcome your feedback and suggestions at metadata@dbtlabs.com.*
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+title: "Querying the Metadata API"
+id: "metadata-querying"
+---
+Accounts on the _Team_ and _Enterprise_ plans can query the dbt Metadata API.
+
+## Authorization
+
+Currently, authorization of requests takes place [using a service token](https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-api/service-tokens). dbt Cloud admin users can generate a Metadata Only service token that is authorized to execute a specific query against the Metadata API.
+
+Once you've created a token, you can use it in the Authorization header of requests to the dbt Cloud Metadata API. Be sure to include the Token prefix in the Authorization header, or the request will fail with a `401 Unauthorized` error. Note that Bearer can be used in place of Token in the Authorization header. Both syntaxes are equivalent.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md]:
```suggestion
Once you've created a token, you can use it in the Authorization header of requests to the dbt Cloud Metadata API. Be sure to include the Token prefix in the Authorization header, or the request will fail with a `401 Unauthorized` error. Note that `Bearer` can be used in place of `Token` in the Authorization header. Both syntaxes are equivalent.
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  --data <query>
+```
+
+The `<query>` body should be a JSON string in the format:
+
+```
+{ “query”: “<query text>” }
+```
+
+Every query will rely on a *jobID*.  You can get the jobID by clicking into the relevant job in dbt Cloud and observing the URL. In this example URL, the jobID would be 917: `https://cloud.getdbt.com/#/accounts/1/projects/665/jobs/917/`
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md:31]:
```suggestion
The currently supported API queries all rely on a *Job ID* to filter metadata requests.  You can get find the relevant ID for your dbt Cloud Job by clicking into the relevant Job in dbt Cloud and inspecting the URL. In this example URL, the Job ID would be 917: `https://cloud.getdbt.com/#/accounts/1/projects/665/jobs/917/`
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+There are several illustrative example queries in this documentation (examples of queries on the Model node, [here](/docs/dbt-cloud/dbt-cloud-api/metadata/schema/metadata-schema-model).
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md:33]:
```suggestion
There are several illustrative example queries in this documentation (examples of queries on the Model node, [here](/docs/dbt-cloud/dbt-cloud-api/metadata/schema/metadata-schema-model)).
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+## GraphiQL
+You can also experiment and run queries directly in the [GraphiQL interface](https://metadata.cloud.getdbt.com/graphiql), which is convenient for exploration. On the right hand side, there is a document explorer where you can see all possible nodes and fields.  Below is an example of what a query looks like in GraphiQL.  Note that you must authenticate via bearer auth with your token.
+
+<Lightbox src="/img/docs/dbt-cloud/GraphiQL.png" title=""/>
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-querying.md:38]: love it!

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-exposure.mdx [hunk 10]
```diff
@@ -0,0 +1,62 @@
+---
+title: "Exposure"
+id: "metadata-schema-exposure"
+---
+
+import { ArgsTable, SchemaTable } from "./schema";
+
+The exposure object allows you to query information about a particular exposure. You can learn more about exposures [here](/docs/building-a-dbt-project/exposures).
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-exposure.mdx:8]: love the linkout to the Exposure docs - great call :)

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx [hunk 11]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Models"
+id: "metadata-schema-models"
+---
+
+import { ArgsTable, SchemaTable } from "./schema";
+
+
+The models object allows you to query information about many models in a given job.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx:9]:
```suggestion
The models object allows you to query information about many models built in a given job.
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+### Arguments
+
+When querying for `models`, the following arguments are available:
+
+<ArgsTable queryName="models" />
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx:15]: can we clarify that the `database`, `schema`, and `identifier` fields are all case-insensitive? I'm ok with that living in the API spec for sure

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+Below we show some illustrative example queries and outline the schema of this models object.
+
+### Example Queries
+As we noted above, database, schema, and identifier are all optional arguments.  This means that with this endpoint, you can:
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-models.mdx:20]:
```suggestion
As noted above, database, schema, and identifier are all optional arguments.  This means that with this endpoint, you can:
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+<ArgsTable queryName="sources" />
+
+Below we show some illustrative example queries and outline the schema of this sources object.
+
+### Example Queries
+As we noted above, database, schema, and identifier are all optional arguments.  This means that with this endpoint, you can:
+- Find a specific model by providing `<database>.<schema>.<identifier>`
+- Find all of the models in a database and/or schema by providing `<database>` and/or `<schema>`
+
+#### Finding models by their database, schema, and identifier
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx]:
```suggestion
#### Finding sources by their database, schema, and identifier
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+```graphql
+{
+  sources(jobId: 123, database:"analytics", schema: "analytics", identifier:"dim_customers") {
+    uniqueId
+  }
+}
+```
+
+#### Finding models by their schema
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx]:
```suggestion
#### Finding sources by their schema
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+When querying for `sources`, the following arguments are available:
+
+<ArgsTable queryName="sources" />
+
+Below we show some illustrative example queries and outline the schema of this sources object.
+
+### Example Queries
+As we noted above, database, schema, and identifier are all optional arguments.  This means that with this endpoint, you can:
+- Find a specific model by providing `<database>.<schema>.<identifier>`
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx]:
```suggestion
- Find a specific source by providing `<database>.<schema>.<identifier>`
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+- Find all of the models in a database and/or schema by providing `<database>` and/or `<schema>`
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema-metadata-sources.mdx]:
```suggestion
- Find all of the sources in a database and/or schema by providing `<database>` and/or `<schema>`
```

File: website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema.jsx [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+            ofType { name description }
+          }
+        }
+      }
+    }
+  }
+}`
+
+
+export const ArgsTable = ({ queryName }) => {
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/dbt-cloud-api/metadata/schema/schema.jsx:32]: this is _so_ rad!

**Review by drewbanin — COMMENTED** [review]
Lots of tiny edits/feedback, but overall looks AMAZING! Let me know if you have any questions about the comments I left in here. Nice work!!

**Review by drewbanin — APPROVED** [review]
:shipit: :shipit: :shipit:

**Pull request merged**